### DISCUSSION TITLE
[form] record data in modifyQueryUsing

### DISCUSSION THREAD

**Discussion by breizhwave** [filamentphp/filament#13169] [General]
Hi, I i am trying to use the current record state to modify a multiple relationship query, as [described here](https://github.com/filamentphp/filament/discussions/9067), except that the state argument returns html of the form, not the record itself. Any idea ?

```
Select::make('position') ->relationship('options', 'title',
                modifyQueryUsing: function (Builder $query, $state) {
                    if (!$state) return $query;
                    print_r($state); die();
                    if (!$state['structure']) return $query;
                        if ($state) {
                            $structure=$state['structure'];
                            if ($structure == "bottom") $where = 'bottom';
                            if ($structure == "top"
                                || $structure == "left"
                                || $structure == "right"
                            ) $where = 'other';
                            if ($where)
                            return    $query->where("category", $where);
                          }
                        return $query;
                    }
                    ,
                ) ->multiple()  ->searchable()->preload(),
```

**Comment by dave-mills** [comment]
I'm surprised if `$state` is returning an html string. It should return the current state of the Select field. 

I think you are seeing the html of the form because the die(); statement is killing the page-load process midway through, so you get the part-rendered page and your print_r($state) inline.

For debugging, I would suggest removing the `preload()`. Then, use `dump($state)` or `dd($state)` instead of `print_r($state); die()`. That way, your debug statements in the function don't interfere with the page load and you'll get the dumped variable in a modal popup when you trigger the select query by typing. 

Note that if you want the entire record, use $record instead of $state. (Docs links: [$state](https://filamentphp.com/docs/3.x/forms/advanced#injecting-the-current-state-of-a-field) vs [$record](https://filamentphp.com/docs/3.x/forms/advanced#injecting-the-current-form-record) )

**Comment by breizhwave** [comment]
O thanks a million for links to [form injection](https://filamentphp.com/docs/3.x/forms/advanced#form-component-utility-injection), it does work better with $record

**Comment by breizhwave** [comment]
New question with code above, I am trying to group select choices by foreign table category field, as described [here](https://github.com/filamentphp/filament/discussions/9471), but does not like the join , should I pluck something ? the following error comes out with the associated SQL , which by the way works in phpmyadmin, but does not return all fields... well ..

SQLSTATE[42000]: Syntax error or access violation: 1055 Expression #1 of SELECT list is not in GROUP BY clause and contains nonaggregated column
```
Select::make('position') ->relationship('options', 'title',
                modifyQueryUsing: function (Builder $query, $record) {
                    if (!$record) return $query;
               
                
                        if ($record) {
                            $structure=$record->structure;
                            if ($structure == "bottom") $where = 'bottom';
                            if ($structure == "top"
                                || $structure == "left"
                                || $structure == "right"
                            ) $where = 'other';
                            if ($where)
                            return    $query->where("category", $where);
                          }
                        return $query;
                    }
                    ,
                ) ->multiple()  ->searchable()->preload(),
```
